Signal/src/UIView+OWS.h

```diff
@@ -60,6 +60,31 @@ CGFloat ScaleFromIPhone5(CGFloat iPhone5Value);
 
 - (void)centerOnSuperview;
 
+#pragma mark - RTL
+
+// For correct right-to-left layout behavior, use "leading" and "trailing",
+// not "left" and "right".
+//
+// NOTE: the margin values are inverted in RTL layouts.
+- (BOOL)isRTL;
+- (NSArray<NSLayoutConstraint *> *)autoPinLeadingAndTrailingToSuperview;
+- (NSLayoutConstraint *)autoPinLeadingToSuperView;
+- (NSLayoutConstraint *)autoPinLeadingToSuperViewWithMargin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinTrailingToSuperView;
+- (NSLayoutConstraint *)autoPinTrailingToSuperViewWithMargin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinLeadingToTrailingOfView:(UIView *)view;
+- (NSLayoutConstraint *)autoPinLeadingToTrailingOfView:(UIView *)view margin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinLeadingToView:(UIView *)view;
+- (NSLayoutConstraint *)autoPinLeadingToView:(UIView *)view margin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinTrailingToView:(UIView *)view;
+- (NSLayoutConstraint *)autoPinTrailingToView:(UIView *)view margin:(CGFloat)margin;
+// Return Right on LTR and Right on RTL.
+- (NSTextAlignment)textAlignmentUnnatural;
+// Leading and trailing anchors honor layout margins.
+// When using a UIView as a "div" to structure layout, we don't want it to have margins.
++ (UIView *)containerView;
+- (void)setHLayoutMargins:(CGFloat)value;
```

Comment:
How is the behavior of this method different from the similar existing API?

e.g. 

```
callStatusLabel.autoPinLeadingToSuperViewWithMargin(contactHMargin)
```

vs. the seemingly similar

```
callStatusLabel.autoPinEdge(toSuperViewEdge:.leading, withInset:contactHMargin)
```

I'm not convinced yet that we need these...  Can you point me to a piece of code where we use these new helpers where it can't be reasonably expressed with the existing autoLayout api? 

I'm not against adding helpers if they solve new problems or are much more readable, but my preference is to rally around a smaller consistent API if reasonable. 

> There's a tricky thing here. I've tried to write these methods so that the "margin" constant makes sense in normal, left-to-right layout. In some cases that requires these methods to invert the actual value .

In fact, do we actually need to handle flipping the constant? When using leading/trailing it seems like autoLayout is mostly doing the right thing already. At least in a couple places I checked, when we manually handle flipping the constant it seems like things go wrong.

e.g. see my comments re: the "close" button on the block banner and (and looks like the same thing with the spinner being off on the "submit code" button during registration.)

It's possible that I just hit a couple of bugs, but I haven't yet seen the case where these are useful.